addons/l10n_in/models/account_invoice.py

```diff
@@ -30,30 +30,41 @@ class AccountMove(models.Model):
     l10n_in_reseller_partner_id = fields.Many2one('res.partner', 'Reseller', domain=[('vat', '!=', False)], help="Only Registered Reseller")
     l10n_in_journal_type = fields.Selection(string="Journal Type", related='journal_id.type')
 
-    @api.depends('partner_id', 'partner_id.l10n_in_gst_treatment', 'state')
+    @api.depends('partner_id', 'partner_id.l10n_in_gst_treatment', 'state', 'partner_shipping_id')
     def _compute_l10n_in_gst_treatment(self):
         indian_invoice = self.filtered(lambda m: m.country_code == 'IN')
         for record in indian_invoice:
             if record.state == 'draft':
-                gst_treatment = record.partner_id.l10n_in_gst_treatment
+                partner = record.partner_shipping_id
+                gst_treatment = partner.l10n_in_gst_treatment
                 if not gst_treatment:
                     gst_treatment = 'unregistered'
-                    if record.partner_id.country_id.code == 'IN' and record.partner_id.vat:
+                    if partner.country_id.code == 'IN' and partner.vat:
                         gst_treatment = 'regular'
-                    elif record.partner_id.country_id and record.partner_id.country_id.code != 'IN':
+                    elif partner.country_id and partner.country_id.code != 'IN':
                         gst_treatment = 'overseas'
                 record.l10n_in_gst_treatment = gst_treatment
         (self - indian_invoice).l10n_in_gst_treatment = False
 
-    @api.depends('partner_id', 'company_id')
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id')
     def _compute_l10n_in_state_id(self):
         for move in self:
             if move.country_code == 'IN' and move.journal_id.type == 'sale':
                 country_code = move.partner_id.country_id.code
                 if country_code == 'IN':
-                    move.l10n_in_state_id = move.partner_id.state_id
+                    if (move.commercial_partner_id == move.partner_shipping_id.commercial_partner_id and
+                    move.partner_shipping_id.state_id.code and
+                    move.partner_shipping_id.country_id.code == 'IN'):
```

Comment:
@masm-odoo I believe you can do something similar as we did above?

I find this a bit complicated here... to read and understand the logic could be more simplified if I m not wrong.

Reply:
As you mentioned, I have optimized it, and now I believe it is much easier to understand.